packages/x-data-grid-premium/src/hooks/features/prompt/api.ts

```
@@ -0,0 +1,40 @@
+import { PromptResponse } from './gridPromptInterfaces';
+
+type Result<T> = { ok: false; message: string } | { ok: true; data: T };
+
+/**
+ * Prompt resolver for the <DataGridPremium /> component.
+ * @param {string} url The URL to send the request to.
+ * @param {string} query The query to be processed.
+ * @param {string} context The prompt context containing necessary information about the current columns definition.
+ * Either use the `context` parameter of the `onPrompt` callback or the return value of the `unstable_getPromptContext()` API method.
+ * @param {string} additionalContext Optional, additional context to make the processing results more accurate.
+ * @returns {Promise<PromptResponse>} The grid state updates to be applied.
+ */
+export function gridDefaultPromptResolver(
+  url: string,
+  query: string,
+  context: string,
+  additionalContext = '',
+) {
+  return fetch(url, {
+    mode: 'cors',
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json',
+    },
+    credentials: 'include',
+    body: JSON.stringify({
+      context,
+      query,
+      additionalContext,
+    }),
+  })
+    .then((result) => result.json())
+    .then((result: Result<PromptResponse>) => {
+      if (result.ok === false) {
```

Would adding a `.catch` in the end be more conventional here? Or it doesn't work in this specific case as no error is thrown?

---

`.catch` is needed together with this
`ok === false` means that the response contains error (like you are not authorized or you have reached the token limit)
`.catch` will catch errors like api service is down completely

---

after refactor, this part will stay the same
if there is an error, by default it will be handled by our assistant panel, so the default resolver just ignores the error and let it pass through

if you make a custom solution, then you can either catch it in your custom resolver or in your custom UI